lib/cn/cn_tree.c

```diff
@@ -1613,34 +1671,35 @@ cn_comp_commit(struct cn_compaction_work *w)
     }
 
     w->cw_err = cndb_record_txstart(w->cw_tree->cndb, 0, CNDB_INVAL_INGESTID, CNDB_INVAL_HORIZON,
-                                    (u16)w->cw_outc, (u16)w->cw_kvset_cnt, &w->cw_cndb_txn);
+                                    w->cw_outc, (u16)w->cw_kvset_cnt, &tx);
```

Comment:
uint16_t while you're here...

Reply:
Or change `cndb_record_txstart` to take `uint` and return `merr(EINVAL)` if a value is out of range.  Casting here will just hide bugs.